types/make.js

```diff
@@ -51,6 +51,8 @@ function generatePrefixed(prefix) {
 				output += '\t' + writeFunction(part, 'name: string', 'void');
 			} else {
 				const type = testType(parts);
+				output += '\t' + writeFunction(part + '<I, E>', 'implementation: Macro<I, E, ' + type + 'Context>, input: I, expected: E');
```

Comment:
This doesn't allow multiple macro's (like in the `test` function in `base.d.ts`. I would define a type alias, after the definition of Macro in `base.d.ts`:
```typescript
type Macros<I, E, T> = Macro<I, E, T> | Macro<I, E, T>[];
```